<html>
<head>
<title>BEAST v2.7.8 Documentation: starbeast3.evolution.speciation.GeneTreeForSpeciesTreeDistribution</title>
<link rel='StyleSheet' href='/tmp/styles.css' type='text/css'>
</head>
<body>
<h1>BEAST v2.7.8 Documentation: starbeast3.evolution.speciation.GeneTreeForSpeciesTreeDistribution</h1>
<p>BEASTObject that performs calculations based on the State.<br/>Probabilistic representation that can produce a log probability for instance for running an MCMC chain.<br/>Calculates probability of gene tree conditioned on a species tree (multi-species coalescent)</p>
<p>Logable: yes, this can be used in a log.</p>
<h2>Inputs:</h2>
<p><a href='#tree'>tree</a>, <a href='#treeIntervals'>treeIntervals</a>, <a href='#speciesTree'>speciesTree</a>, <a href='#ploidy'>ploidy</a>, <a href='#sampling'>sampling</a>, <a href='#speciesTreePrior'>speciesTreePrior</a>, <a href='#populationModel'>populationModel</a>, <a href='#taxonset'>taxonset</a </p>
<p>&nbsp</p><table id='tree' border='1px' width='90%'>
<caption>tree</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: beast.base.evolution.tree.TreeInterface</td></tr></thead>
<tr><td>tree over which to calculate a prior or likelihood</td></tr>
<tr><td>
Required input</td></tr>
</table>
<p>&nbsp</p><table id='treeIntervals' border='1px' width='90%'>
<caption>treeIntervals</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: <a href='beast.base.evolution.tree.TreeIntervals.html'>beast.base.evolution.tree.TreeIntervals</a></td></tr></thead>
<tr><td>Intervals for a phylogenetic beast tree</td></tr>
<tr><td>
Either this, or tree needs to be specified</td></tr>
</table>
<p>&nbsp</p><table id='speciesTree' border='1px' width='90%'>
<caption>speciesTree</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: <a href='starbeast3.tree.SpeciesTree.html'>starbeast3.tree.SpeciesTree</a></td></tr></thead>
<tr><td>species tree containing the associated gene tree</td></tr>
<tr><td>
Required input</td></tr>
</table>
<p>&nbsp</p><table id='ploidy' border='1px' width='90%'>
<caption>ploidy</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: java.lang.Double</td></tr></thead>
<tr><td>ploidy (copy number) for this gene, typically a whole number or half (default 2 for autosomal_nuclear)</td></tr>
<tr><td>
Optional input. Default: 2.0</td></tr>
</table>
<p>&nbsp</p><table id='sampling' border='1px' width='90%'>
<caption>sampling</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: java.lang.Boolean</td></tr></thead>
<tr><td>Set to true if using this class for simulating sequences</td></tr>
<tr><td>
Optional input. Default: false</td></tr>
</table>
<p>&nbsp</p><table id='speciesTreePrior' border='1px' width='90%'>
<caption>speciesTreePrior</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: <a href='starbeast3.evolution.speciation.SpeciesTreePrior.html'>starbeast3.evolution.speciation.SpeciesTreePrior</a></td></tr></thead>
<tr><td>defines population function and its parameters</td></tr>
<tr><td>
Optional input</td></tr>
</table>
<p>&nbsp</p><table id='populationModel' border='1px' width='90%'>
<caption>populationModel</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: starbeast3.evolution.speciation.PopulationModel</td></tr></thead>
<tr><td>Population model used to infer the multispecies coalescent probability for this gene</td></tr>
<tr><td>
Optional input</td></tr>
</table>
<p>&nbsp</p><table id='taxonset' border='1px' width='90%'>
<caption>taxonset</caption>
<thead><tr bgcolor='#AAAAAA'><td>type: <a href='beast.base.evolution.alignment.TaxonSet.html'>beast.base.evolution.alignment.TaxonSet</a></td></tr></thead>
<tr><td>set of taxa mapping lineages to species</td></tr>
<tr><td>
Optional input</td></tr>
</table>
</body>
</html>
