---
layout: post
title: All about starting trees 
tags: [beast-2,starting-tree,v2-1-3,xml]
---

	

		
<p style="color:gray">28 July 2014, updated 20 Mar 2018 by <a href='mailto:r.bouckaert@auckland.ac.nz'>Remco Bouckaert</a></p>
<p>Before starting an MCMC run, BEAST uses <tt>State Node</tt> initialisers to set up the starting trees (and other parameters). Often a good starting tree is already available from another analysis (e.g. a maximum likelihood tree). Though it is against the spirit of MCMC, which guarantees any random starting state will lead to convergence, a good starting tree can help with large analysis to speed up convergence quite a bit. Of course another reason to specify a starting tree is if you want to do an analysis using just a single known tree topology.</p>
<p>We start with standard analysis, but see below for *BEAST and StarBeast2 analyses.</p>

<h2>Standard analysis</h2>

<p>For a standard analysis, there are a number of ways to specify a starting tree </p>
<ul>
<li>a random tree based on the coalescent</li>
<li>a tree in Newick format</li>
<li>a cluster tree, e.g. through UPGMA or neighbour joining</li>
<li>a simple random tree</li>
</ul>
The easiest way to set up a starting tree is in BEAUti; click the menu <code>View/Starting tree panel</code>.

<center><img width="75%" src="/images/BEAUTIViewStartinTree.png"></center>

From the drop-down box you can select the type of tree you want to start with.
<center><img width="75%" src="/images/BEAUTIStartinTreeRandom.png"></center>
A random tree is a tree drawn from a coalescent process, so it uses a population function to guide that process. When there are time calibrations in the tree, it is possible the tree will not conform to these calibrations. You can increase or decrease the population size to get bigger or smaller (respectively) trees. Alternatively, you can give a target root height and the tree will be scaled to match that height if constraints allow this.


<center><img width="75%" src="/images/BEAUTIStartinTreeCluster.png"></center>

A rough and ready tree can be obtained using clustering, which can be expected to be fairly close to trees from the posterior (at least for the topology). BEAST has a number of cluster algorithms implemented, including UPGMA and neighbour joining. Note that using such a tree is non-random, so using it  as starting tree means the starting tree will always be the same one. The <code>ClusterTree</code> attempts to take monophyletic constraints in account.

<center><img width="75%" src="/images/BEAUTIStartinTreeNewick.png"></center>

When you have another source for a starting tree and have it in Newick format, you can select the Newick tree and paste it in the <code>Nnewick</code> entry. Note that if you only have the topology and have timing constraints (i.e., MRCA priors) branch lengths are set to 0.001. Furthermore, if the <code>Adjust Tip Heights</code> flag is set, then tip heights will be set to the height specified in the tip-dating panel (or at zero, if nothing is specified), adjusting branch lengths in the process.
You can scale the tree with a fixed factor using the <code>scale</code> entry.


<center><img width="75%" src="/images/BEAUTIStartinTreeSimpleRandom.png"></center>
<p>The <code>RandomTree</code> will not always be able to deal with all combinations of calibrations. The <code>SimpleRandomTree</code> available in the <code><a href="https://github.com/BEAST2-Dev/BEASTLabs">BEASTLabs</a></code> package is more robust. However, it is not based on the coalescent. For BEAST 2.5, a template will be available so you can choose this in BEAUti. For v2.4, it is possible to set edit the XML (see below).


<h2>Editing the XML for a Standard Analysis</h2>

<p>By default, BEAUti uses a RandomTree state-node initialiser to generate a starting tree. If you want to replace it with a Newick tree or cluster tree, first you need to remove (or comment out) the RandomTree element from the XML. For a tree in partition <tt>XYZ26</tt>, just search for <tt>RandomTree</tt> and remove the XML fragment with id <tt>RandomTree.t:XYZ26</tt>. It should look something like this:</p>
<p></p>
			
			{% highlight xml %}<init estimate="false" id="RandomTree.t:XYZ26" initial="@Tree.t:XYZ26" spec="beast.evolution.tree.RandomTree" taxa="@XYZ26">
    <populationModel id="ConstantPopulation0.t:XYZ26" spec="ConstantPopulation">
        <parameter id="randomPopSize.t:XYZ26" name="popSize">1.0</parameter>
    </populationModel>
</init>{% endhighlight %}			
			
		

<p></p>
<p>Before removing, make note of the initial attribute (<tt>@Tree.t:XYZ26</tt>> here) and taxa attribute (<tt>@XYZ26</tt> here).</p>
<h3>Newick starting tree</h3>
<p>After removing the Random tree, in the same place, add a <tt>TreeParser</tt> using the following fragment</p>
<p></p>

		
		
			
			
			
			{% highlight xml %}<init spec='beast.util.TreeParser' id='NewickTree.t:XYZ26' 
  initial="@Tree.t:XYZ26" taxa='@XYZ26' IsLabelledNewick="true" 
  newick="((your,(tree,goes)),here)"/>{% endhighlight %}			
			
		

<p></p>
<p>The attributes that you want to specify are</p>
<ul>
<li><tt>id="NewickTree.t:XYZ26"</tt> the ID should be unique, so something like <tt>NewickTree.t:</tt> plus the name of the tree would be suitable.
<li><tt>initial="@Tree.t:XYZ26"</tt> this refers to the tree being initialised. It should be the same as the initial attribute of the RandomTree that was removed.
<li><tt>taxa="@XYZ26"</tt> this refers to the alignment and ensures that taxa in the tree are lined with those in the alignment. For BEAST v2.1.x, this must be specified to prevent starting with a mislabelled tree
<li><tt>newick="((your,(tree,goes)),here)"</tt> obviously needs to be replaced with your own tree in Newick format.
<li>
</ul>
<p>This is all you need to know to set up a starting tree. There are a few optional attributes you can use to make life a bit easier:</p>
<ul>
<li><tt>adjustTipHeights</tt> is true by default, which means tips of the tree will be set to zero, or if tip dates are specified to these particular tip dates. By setting to <tt>adjustTipHeights="false"</tt>, tips will be initialised by the heights in the Newick tree.
<li> if <tt>adjustTipHeights="false"</tt> then <tt>threshold</tt> specifies threshold under which node heights (derived from lengths) are set to zero. This helps when there are numeric issues with adding the lengths.
<li><tt>scale</tt> scale used to multiply internal node heights during parsing. Useful for importing starting from external programs, for instance, RaxML tree rooted using Path-o-gen.
<li><tt>IsLabelledNewick="true"</tt> Is the Newick tree labelled (alternatively contains node numbers)?
<li>If <tt>sLabelledNewick="false"</tt> then &#8220;<tt>offset="1"</tt>  is the lowest taxa number. The default=1 but 0 is common as well. Taxa numbers are as they are ordered in the alignment referred to with the <tt>taxa</tt> attribute.
<li>
</ul>
<h3>Cluster tree</h3>
<p>After removing the Random tree, in the same place, add a <tt>ClusterTree</tt> with the following fragment</p>
<p></p>

		
		
			
			
			
			{% highlight xml %}<init spec='beast.util.ClusterTree' id='UPGMATree.t:XYZ26' 
  initial="@Tree.t:XYZ26" clusterType='upgma' 
  taxa='@XYZ26'/>{% endhighlight %}			
			
		

<p></p>
<p>The attributes that you need to specify are <tt>id='UPGMATree.t:XYZ26'</tt>, <tt>initial="@Tree.t:XYZ26"</tt> and <tt>taxa="@XYZ26"</tt>, which are as for Newick tree (see above).</p>
<p>You can use different clustering algorithms by specifying the <tt>clusterType="upgma"</tt> attribute.</p>
<table>
<tbody>
<tr>
<th width='50%'>clusterType</th>
<th width='50%'>Description</th>
</tr>
<tr>
<td>single</td>
<td>single link</td>
</tr>
<tr>
<td>complete</td>
<td>complete link</td>
</tr>
<tr>
<td>upgma or average</td>
<td>UPGMA=average link</td>
</tr>
<tr>
<td>mean</td>
<td>mean link</td>
</tr>
<tr>
<td>centroid</td>
<td>centroid</td>
</tr>
<tr>
<td>ward</td>
<td>Ward&#8217;s method</td>
</tr>
<tr>
<td>adjcomplete</td>
<td>adjusted complete link</td>
</tr>
<tr>
<td>neighborjoining</td>
<td>neighborjoining</td>
</tr>
<tr>
<td>neighborjoining2</td>
<td>neighborjoining2 &#8211; corrects tree for tip data, unlike plain neighborjoining</td>
</tr>
</tbody>
</table>

<h3>Simple Random Tree</h3>

For using a <code>SimpleRandomTree</code> you need to install the <code><a href="https://github.com/BEAST2-Dev/BEASTLabs">BEASTLabs</a></code> package. Instead of removing the <code>Random</code> tree, only remove the population model and in the spec-attribute, replace <code>RandomTree</code> with <code>SimpleRandomTree</code>:


			{% highlight xml %}<init estimate="false" id="RandomTree.t:XYZ26" initial="@Tree.t:XYZ26" spec="beast.evolution.tree.SimpleRandomTree" taxa="@XYZ26">
    <!-- comment out (or delete) population model
    <populationModel id="ConstantPopulation0.t:XYZ26" spec="ConstantPopulation">
        <parameter id="randomPopSize.t:XYZ26" name="popSize">1.0</parameter>
    </populationModel>
    -->
</init>{% endhighlight %}			


<h2>*BEAST analysis</h2>
<p>For a *BEAST analysis, the default in BEAUti is to generate a UPGMA tree for the species tree, then generate UPGMA trees for each of the gene trees and fit them inside the species tree. In previous incarnations of BEAST, a random tree was generated for species as well as for gene trees, but gene tree branches ending in leaves were lengthened such that each first coalescence of a gene tree was above the root of the species tree. Though this ensures that the gene trees fit inside the species tree, it leads to long burn-in times.</p>
<p>If you want to use a different starting tree for the species tree, first move (not remove!) the <tt>StarBeastStartState</tt> element generated by BEAUti to just before the run-element with <tt>id="mcmc"</tt>. BEAUti will have generated some elements inside the start-state (such as the species-tree-prior) that we need elsewhere in the analysis. By moving the init element outside the run element, it will not activate the <tt>StarBeastStartState</tt>.</p>
<p>The <tt>StarBeastStartState</tt> looks something like this:</p>
<p></p>

		
		
			
			
			
			{% highlight xml %}<init birthRate="@birthRate.t:Species" estimate="false" id="SBI" popMean="@popMean" spec="beast.evolution.speciation.StarBeastStartState" speciesTree="@Tree.t:Species">
    <tree idref="Tree.t:29" name="gene"/>
    <tree idref="Tree.t:26" name="gene"/>
    <speciesTreePrior bottomPopSize="@popSize" gammaParameter="@popMean" id="SpeciesTreePopSize.Species" spec="beast.evolution.speciation.SpeciesTreePrior" taxonset="@taxonsuperset" tree="@Tree.t:Species"/>
</init>{% endhighlight %}			
			
		

<p></p>
<p>For every gene tree there will be a tree entry in the <tt>StarBeastStartState</tt>. To replace the species tree with a Newick tree, add the following fragment (at the place where the <tt>StarBeastStartState</tt> was):</p>
<p></p>

	
		
			
			
			
			{% highlight xml %}<init spec='beast.util.TreeParser' id='NewickTree.t:Species' 
   initial="@Tree.t:Species" taxonset='@taxonsuperset' 
   IsLabelledNewick="true" 
   newick="((your,(tree,goes)),here)"/>{% endhighlight %}			
			
		

<p></p>
<p>The only thing to set up is the Newick using the <tt>newick</tt> attribute &#8212; all other attributes should be as they are (it is using the species tree, which is identified by the Species partition in every *BEAST analysis generated by BEAUti).</p>
<p>For each of the gene trees, you need to specify a tree as well. You can specify them using a Newick tree as for the Standard analysis (see above). Alternatively, you can use a random tree that is placed &#8216;above&#8217; the species tree. To do this, you add for the first gene tree (just below the new Species tree start tree):</p>
<p></p>

		
		
			
			
			
			{% highlight xml %}<init spec='beast.evolution.speciation.RandomGeneTree' 
      id='randomGeneTree1' initial='@Tree.t:26' 
      speciesTree='@Tree.t:Species' taxa='@26'>
  <populationModel id='popmodel' spec='ConstantPopulation' popSize="1.0"/>
</init>{% endhighlight %}			
			
		

<p></p>
<p>where you replace the <tt>initial='@Tree.t:26'</tt> and <tt>taxa="@26"</tt> attributes to match your tree (instead of 26, as in this example). For every subsequent gene tree, you add (below the first gene start tree):</p>
<p></p>

		
		
			
			
			
			{% highlight xml %}<init spec='beast.evolution.speciation.RandomGeneTree' 
  id='randomGeneTree2' initial='@Tree.t:29' 
  speciesTree='@Tree.t:Species' taxa='@29'   
  populationModel='@popmodel'/>{% endhighlight %}			
			
		

<p></p>
<p>where you replace the <tt>initial='@Tree.t:29'</tt> and <tt>taxa="@29"</tt> attributes to match your tree (instead of 29) and make sure the id is unique.</p>
	
<h2>StarBEAST2 analysis</h2>

StarBEAST2 analysis require the <code><a href="https://github.com/genomescale/starbeast2">StarBEAST2</a></code> package. 
Setting up starting trees is very similar to setting them up for a *BEAST analysis, but
these analyses use a custom species tree, and a custom <code>StarBeastInitialiser</code> with <code>id="SBI"</code>. To change the starting trees, you should remove the element with <code>id="SBI"</code>, and follow instructions for *BEAST starting trees above.
