---
layout: post
title: Species Delimitation with BEAST 
tags: [beast-2,bfd,dissect,species-delimitation,stacey]
---

	

		
		<p style="color:gray">2 June 2015 by <a href='mailto:remco@cs.auckland.ac.nz'>Remco Bouckaert</a></p>
<p>A few weeks back I attended the conference on <a href="http://cba.anu.edu.au/news-events/species-delimitation-age-genomics">Species delimitation in the age of genomics</a> at ANU, which made me realise there is quite an interest in the topic. So, here a quick review of methods for species delimitation available in BEAST. The main methods are</p>
<ul>
<li>Bayes factor delimitation (BFD)</li>
<li>Threshold based methods DISSECT/STACEY</li>
</ul>
<h2>Bayes factor delimitation</h2>
<p>BFD is based on the multi-species coalescent where there two or more scenarios for species assignments. These scenarios can differ in that taxa for different species can be merged, split or you can even test whether a lineage fits better with one species than another.</p>
<p>The multi-species coalescent can be based on *BEAST or SNAPP, depending on the kind of data that you have: gene sequences for *BEAST and SNP or AFLP data for SNAPP. The idea of BFD is based on a fundamental method of Bayesian methods, which is comparing models based on their marginal likelihoods. Note that this is different from the likelihood typically shown in Tracer; the marginal likelihood is the likelihood marginalised over all parameters in the model.</p>

		
		
			
			
			
			{% highlight xml %}              prior x likelihood
posterior =  --------------------
              marginal likelihood{% endhighlight %}			
			
		

<p>The marginal likelihood is most reliably calculated using a stepping stone analysis, though this can be quite tedious to estimate since it is rather computational intensive. There are other methods for model fit, like AICM, that are less computational intensive, but these tend to be less reliable (as outlined by <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0106990">Ayden et al, PloS one, 2014</a>).</p>
<p>For each of the scenarios, you set up a *BEAST or SNAPP analysis with a different species assignment, and estimate the marginal likelihood for each of these scenarios. To do this in BEAST, you need the MODEL_SELECTION package. There are more details <a href="http://blog.beast2.org/2014/07/14/path-sampling-with-a-gui/">here</a> on how to set up a stepping stone/path sampling analysis.</p>
<p>Once you have the marginal likelihoods for each of your scenarios, the Bayes factor comparing say scenario A and B is just the difference between the marginal likelihood estimates.</p>
<h2>Threshold based methods DISSECT/STACEY</h2>
<p>Threshold based methods are based on setting a level epsilon and declare any split in the species tree that is below that value to be within a species, while any split above the threshold is deemed to represent the birth of a new species. The benefit of this method is that it can be performed during an MCMC run, so it does not require a stepping stone analysis. Also, it does not require setting up different scenarios. However, it does require setting a rather arbitrary threshold, though at the meeting at the ANU conference it was argued by various speakers that defining species is to some extent a social construct, which involves some subjective criteria. So, perhaps this is not such a big deal. <b>update</b>: Graham Jones clarified that the threshold is an approximation to zero without any biological meaning, so should not be interpreted as a social construct but its choice is based on a balance of accuracy and speed.</p>
<p>The DISSECT and STACEY packages allows you to do threshold based species delimitation. STACEY is *BEAST on steroids, with DISSECT as species delimitation method. It integrates out population sizes for the branches, which helps convergence though it means that population size information is lost. Furthermore, it has a number of MCMC proposals that help in mixing; a *BEAST analysis can easily get stuck in an area of tree space that forms a local optimum where the species tree gets locked into a topology because one or more of the gene trees have a good fit to the data preventing the species tree to move to alternative topologies. STACEY offers three extra proposals on top of the standard *BEAST proposals that help to get out of these local optima.</p>
<p>A STACEY analysis can be set up in BEAUti, but DISSECT requires a bit more handwork in editing the XML. For STACEY, you need to select the STACEY template under the File/Templates menu to get started. Both need a bit of fiddling with the XML to get going &#8212; see documentation for these packages to work out the details.</p>
<p>After running a DISSECT or STACEY XML file through BEAST, you can use the <a href="http://indriid.com/software.html">SpeciesDelimitationAnalyser</a> to process the log files and find out the distribution over species assignments.</p>
<p>Threshold methods have been tested for *BEAST, though should work without too much hassle with SNAPP. The prior for SNAPP needs to be replaced with the</p>
<h2>Links</h2>
<p><a href="http://sysbio.oxfordjournals.org/content/63/2/119.short">BFD paper</a> for *BEAST.</p>
<p><a href="http://sysbio.oxfordjournals.org/content/63/4/534.short">BFD* paper</a> for SNAPP.</p>
<p><a href="https://github.com/CompEvol/CBAN/releases/download/v2.1.2/BFD-tutorial.zip">Tutorial</a> for BFD*, with example data.</p>
<p><a href="http://bioinformatics.oxfordjournals.org/content/early/2014/12/10/bioinformatics.btu770.full.pdf+html">DISSECT</a> paper, also at <a href="http://biorxiv.org/content/early/2014/03/03/003178">bioRxiv</a>.</p>
<p><a href="http://indriid.com/workingnotes2015.html">STACEY</a> info and <a href="http://biorxiv.org/content/early/2015/03/22/010199">preprint</a>.</p>
	
	
