---
layout: post
title: Model adequacy in phylogenetics -- introducing the TreeModelAdequacy package for BEAST2
tags: []
---
<p style="color:gray">24 July 2018 by <a href='maito:sebastian.duchene@unimelb.edu.au'>Sebastian Duchene</a> and <a href='mailto:remco@cs.auckland.ac.nz'>Remco Bouckaert</a></p>

<p>In this post we provide a brief introduction into model adequacy. We describe some theoretical background and the key differences between model adequacy and model selection. This was motivated by the recent release of our BEAST2 package, <a href="https://github.com/sebastianduchene/tree_model_adequacy">TreeModelAdequacy</a>, to assess the adequacy of the tree model (sometimes known as the `tree prior') in Bayesian phylogenetic analyses. Model adequacy differs from model selection in that it can assess the absolute, rather than relative, performance of the model. This is useful to assess whether the resulting inferences are reliable and to determine whether key features of the data could have been generated by the model in question. An upcoming post will be a tutorial for using <tt>TreeModelAdequacy</tt> in a virus data set. 

<h2>Model selection in phylogenetics</h2>

<p>Bayesian phylogenetic methods have increased the range of models for evolutionary analyses. For example, a typical analysis in BEAST2 involves a substitution model, a clock model, and a tree prior, to describe sequence evolution, evolutionary rate variation, and the branching pattern, respectively. We will refer to this ensemble of models as the complete `Bayesian phylogenetic model' (<a href="#duPlessis2015">du Plessis and Stadler 2015</a>; <a href="#Bromham2018">Bromham et al. 2018</a>). Model choice is an important aspect of phylogenetic analyses because choosing an incorrect model can lead to misleading inferences. 

<p>The impact of the substitution model in estimates of phylogenetic trees was recognised since statistical phylogenetic methods were proposed (<a href="#Sullivan2005">Sullivan and Joyce 2005</a>). In general, using an incorrect substitution model can result in incorrect inferences of tree topologies and branch lengths. For this reason, substitution model selection became standard practice in phylogenetic analyses. The typical approach to selecting models consists in considering a pool of models (e.g., a subset of the GTR family) and ranking them with respect to some statistical criteria, such as the Akaike Information Criterion (AIC), the Bayesian Information Criterion (BIC). The AIC and BIC consider the maximum likelihood of the model and penalise the number of parameters to compute a score. Alternatively, models can also be compared based on a likelihood-ratio test, but unlike the AIC and BIC the models being compared must be nested. Popular programs for substitution model selection include JModeltest (<a href="#Posada2008">Posada 2008</a>). Bayesian approaches to model selection include those based on stepping-stone, path-sampling (<a href="#Xie2011">Xie et al. 2011</a>; <a href="#Baele2013">Baele et al. 2013</a>, <a href="#Baele2016">2016</a>), and nested-sampling (<a href="#Maturana2018">Maturana et al. 2018</a>). Their aim is to approximate the marginal likelihood of a given model that naturally penalises excess parameters and can be used to effectively select the optimal model. 

<p>More recently, there has been a rise in model averaging methods (<a href="#Posada2004">Posada and Buckley 2004</a>; <a href="#Baele2013">Baele et al. 2013</a>; <a href="#Bouckaert2017">Bouckaert and Drummond 2017</a>). The aim of model averaging is to obtain weighted parameter estimates across a range of models. In maximum likelihood this can be done by averaging branch lengths and computing consensus trees over a range of models, using the AIC or BIC scores as weights. In Bayesian phylogenetics, the substitution model can be treated as an unknown `parameter' during the MCMC. The frequency with which models are sampled corresponds to their posterior probability. As such, the best model is that which was most frequently sampled, also known as the Maximum `a posteriori' model (MAP). However, the greatest benefit of model averaging is that it is not necessary to condition the inferences on a single model. 

<h2>Model adequacy</h2>

<p>In spite of the importance of selecting an appropriate model, model selection methods only assess the relative performance of a pool of models. A salient problem about this approach is that it does not assess absolute model performance (<a href="#Brown208">Brown and Thomson 2018</a>). Consider a situation where all of the models considered poorly describe the data. Model selection methods will still select one of the models considered, and the researcher will have no information as to whether it captures the main features of the data. In the case of substitution models a common problem is compositional heterogeneity, where very divergent lineages have evolved similar base compositions and most phylogenetic methods may incorrectly cluster them together. Conducting model selection under models of the GTR family only will not reveal that the data are inadequately described by all of these models. 

<p>The purpose of model adequacy is to assess whether a given model could have generated the data at hand (<a href="#Gelman1996">Gelman et al. 1996</a>). In practice, this can be done by fitting a model to the data and obtaining estimates for all parameters. These parameter estimates are used to simulate a number of data sets, which represent hypothetical future observations if the model were correct. We consider that a model adequately describes the empirical data if the simulations and empirical data are very similar according to a test statistic. Designing test statistics is a difficult task and is the crux of model adequacy. 

<h2>Test statistics for model adequacy</h2>

<p>There has been substantial work to determine test statistics for substitution models (<a href="#Huelsenbeck2001">Huelsenbeck et al. 2001</a>; <a href="#Bollback2002">Bollback 2002</a>; <a href="#Foster2004">Foster 2004</a>; <a href="#Ripplinger2010">Ripplinger and Sullivan 2010</a>; <a href="#Brown2014">Brown 2014</a>; <a href="#Lewis2014">Lewis et al. 2014</a>; <a href="#Duchêne2015">Duchêne et al. 2015</a>; <a href="#Duchene2016">Duchene et al. 2016</a>; <a href="#Duchêne2018a">Duchêne et al. 2018a</a>), with several available implementations (<a href="#Höhna">Höhna et al. 2017</a>; <a href="#Duchêne2018b">Duchêne et al. 2018b</a>). Common test statistics include the multinomial likelihood and a &chi;<sup>2</sup> test of compositional heterogeneity. However, other aspects of the complete Bayesian phylogenetic model have been received less attention, notably those that describe the branching process in phylogenetic trees (<a href="#Drummond2008">Drummond and Suchard 2008</a>), sometimes known as the `tree prior' (<a href="#duPlessis2015">du Plessis and Stadler 2015</a>). This is especially relevant in the field of phylodynamics, where a large number of inferences can be drawn from the branching process, such as transmission rates of infectious pathogens (<a href="#Pybus2003">Pybus et al. 2003</a>; <a href="#Frost">Frost and Volz 2010</a>; <a href="#Stadler2012">Stadler et al. 2012</a>). In fact, recently there has been substantial effort in the development of these models to include epidemiological models (<a href="#Kühnert2014">Kühnert et al. 2014</a>, <a href="#Kühnert2016</a>;">2016</a>; <a href="#Vaughan">Vaughan et al. 2017</a>), describe population structure (<a href="#Müller2017">Müller et al. 2017</a>).

<p>We developed <a href="https://github.com/sebastianduchene/tree_model_adequacy"><tt>TreeModelAdequacy</tt></a> to assess the adequacy of tree models (<a href="#Duchene2018a">Duchene et al. 2018a</a>). This package takes a phylogenetic tree as input to infer parameters of the tree model, such as the birth and death rates of a birth-death model. We will refer to this tree as the `empirical tree' and the parameters of the model as &eta;. We estimate the posterior distribution of &eta;. Then we draw a number of random samples, such as 100, from the posterior of &eta; to simulate phylogenetic trees (<a href="#Fig1">Fig 1</a>). These phylogenetic trees are known as posterior predictive simulations. Finally, we calculate a number of test statistics for the empirical tree and the posterior predictive simulations. The distribution of a test statistic for the posterior predictive simulations is known as a posterior predictive distribution.

<img id="Fig1" src="/images/TreeModelAdequacy.png"/>

<b>Fig 1.</b> Posterior predictive simulation framework implemented in the <tt>TreeModelAdequacy</tt> package. Step (1) consists in Bayesian analysis in BEAST under model <tt>M</tt> to estimate the posterior distribution of parameters &eta;, shown with the arrow and posterior density in orange. In step (2) samples from the posterior are drawn to simulate phylogenetic trees, known as posterior predictive simulations, using MASTER as shown by the green arrow. In step (3) the posterior predictive simulations are analysed in <tt>TreeStat</tt> to generate the posterior predictive distribution of test statistic <tt>T<sub>a</sub></tt>, shown by the blue arrow and probability density. Finally, T<sub>a</sub> is also computed for the tree from the empirical data using <tt>TreeStat</tt>, shown by the red arrow, to calculate a posterior predictive probability (<tt>p<sub>B</sub></tt>). Test statistics and <tt>p<sub>B</sub></tt> values can also be computed for trees generated in other programs using <tt>TreeModelAdequacyAnalyser</tt>, given that the tree from the empirical data and the posterior predictive simulations are provided.

<p>Some examples of test statistics that are informative to assess the tree model are the root-node height, or the ratio of internal to external branch lengths, but there are many more available in <tt>TreeModelAdequacy</tt>. We determine where the value of the test statistic for the empirical tree falls with respect to the posterior predictive distribution by calculating a posterior predictive p-value for each test statistic, known as <tt>p<sub>B</sub></tt> to distinguish it from the frequentist p-value (<a href="#Gelman2014">Gelman et al. 2014</a>; <a href="#Duchene2018a">Duchene et al. 2018a</a>). We consider that a given aspect of the data (e.g., root-node height) is adequately described by the model if <tt>p<sub>B</sub></tt> is between 0.025 and 0.975 (i.e., within the 95% credible interval of the posterior predictive distribution). A basic tutorial for <tt>TreeModelAdequacy</tt> is available <a href="https://github.com/sebastianduchene/tree_model_adequacy/wiki/How-to-use-Tree-Model-Adequacy">here</a>.

<p>An important aspect of this approach is that it assumes a single tree. The tree can be a summary tree from BEAST or it could be estimated using a different method. If the tree has been estimated using BEAST it is important to verify that it has strong branch support and that the sequence data used to obtain are sufficiently informative (<a href="#Duchene2018b">Duchene et al. 2018b</a>). In practice, this can be verified by comparing the prior and posterior distributions. 


<h2>References</h2>

<p id="Baele2016">Baele G., Lemey P., Suchard M.A. 2016. Genealogical working distributions for Bayesian model testing with phylogenetic uncertainty. Syst. Biol. 65:250–264.
<p id="Baele2013">Baele G., Li W.L.S., Drummond A.J., Suchard M.A., Lemey P. 2013. Accurate model selection of relaxed molecular clocks in bayesian phylogenetics. Mol. Biol. Evol. 30:239–243.
<p id="Bollback2002">Bollback J.P. 2002. Bayesian model adequacy and choice in phylogenetics. Mol. Biol. Evol. 19:1171–1180.
<p id="Bouckaert2017">Bouckaert R., Drummond A.J. 2017. bModelTest: Bayesian phylogenetic site model averaging and model comparison. BMC Evol. Biol. 17:42.
<p id="Bromham2018">Bromham L., Duchêne S., Hua X., Ritchie A.M., Duchêne D.A., Ho S.Y.W. 2018. Bayesian molecular dating: Opening up the black box. Biol. Rev. 93:1165–1191.
<p id="Brown2014">Brown J.M. 2014. Predictive approaches to assessing the fit of evolutionary models. Syst. Biol. 63:289–292.
<p id="Brown2018">Brown J.M., Thomson R.C. 2018. Evaluating Model Performance in Evolutionary Biology. Annu. Rev. Ecol. Evol. Syst. In press.
<p id="Drummond2008">Drummond A.J., Suchard M.A. 2008. Fully Bayesian tests of neutrality using genealogical summary statistics. BMC Genet. 9:68.
<p id="Duchêne2018a">Duchêne D.A., Duchêne S., Ho S.Y.W. 2018a. Differences in Performance among Test Statistics for Assessing Phylogenomic Model Adequacy. Genome Biol. Evol. 10:1375–1388.
<p id="Duchêne2018b">Duchêne D.A., Duchêne S., Ho S.Y.W., Kelso J. 2018b. PhyloMAd: Efficient assessment of phylogenomic model adequacy. Bioinformatics. 34:2300–2301.
<p id="Duchêne2015">Duchêne D.A., Duchêne S., Holmes E.C., Ho S.Y.W. 2015. Evaluating the adequacy of molecular clock models using posterior predictive simulations. Mol. Biol. Evol. 32.
<p id="Duchene2018a">Duchene S., Bouckaert R., Duchene D.A., Stadler T., Drummond A.J. 2018a. Phylodynamic model adequacy using posterior predictive simulations. Syst. Biol. In press.
<p id="Duchene2018a">Duchene S., Duchene D., Geoghegan J., Dyson Z.A., Hawkey J., Holt K.E. 2018b. Inferring demographic parameters in bacterial genomic data using Bayesian and hybrid phylogenetic methods. BMC Evol. Biol. 18:95.
<p id="Duchene2016">Duchene S., Di Giallonardo F., Holmes E.C. 2016. Substitution model adequacy and assessing the reliability of estimates of virus evolutionary rates and time scales. Mol. Biol. Evol. 33.
<p id="Foster2004">Foster P.G. 2004. Modeling compositional heterogeneity. Syst. Biol. 53:485–495.
<p id="Frost2010">Frost S.D.W., Volz E.M. 2010. Viral phylodynamics and the search for an `effective number of infections.' Philos. Trans. R. Soc. London B Biol. Sci. 365:1879–1890.
<p id="Gelman2014">Gelman A., Carlin J.B., Stern H.S., Dunson D.B., Vehtari A., Rubin D.B. 2014. Model checking. Bayesian data analysis. Boca Raton, Florida: CRC press Boca Raton, FL. p. 141–163.
<p id="Gelman1996">Gelman A., Meng X.-L., Stern H. 1996. Posterior predictive assessment of model fitness via realized discrepancies. Stat. Sin.:733–760.
<p id="Höhna2017">Höhna S., Coghill L.M., Mount G.G., Thomson R.C., Brown J.M. 2017. P3: Phylogenetic Posterior Prediction in RevBayes. Mol. Biol. Evol. 35:1028–1034.
<p id="Huelsenbeck2001">Huelsenbeck J.P., Ronquist F., Nielsen R., Bollback J.P. 2001. Bayesian inference of phylogeny and its impact on evolutionary biology. Science. 294:2310–2314.
<p id="Kühnert2014">Kühnert D., Stadler T., Vaughan T.G., Drummond A.J. 2014. Simultaneous reconstruction of evolutionary history and epidemiological dynamics from viral sequences with the birth–death SIR model. J. R. Soc. Interface. 11:20131106.
<p id="Kühnert2016">Kühnert D., Stadler T., Vaughan T.G., Drummond A.J. 2016. Phylodynamics with migration: A computational framework to quantify population structure from genomic data. Mol. Biol. Evol. 33:2102–2116.
<p id="Lewis204">Lewis P.O., Xie W., Chen M.-H., Fan Y., Kuo L. 2014. Posterior predictive Bayesian phylogenetic model selection. Syst. Biol. 63:309–321.
<p id="Maturana2018">Maturana P., Brewer B.J., Klaere S., Bouckaert R. 2018. Model selection and parameter inference in phylogenetics using Nested Sampling. Syst. Biol. In press.
<p id="Müller2017">Müller N.F., Rasmussen D.A., Stadler T. 2017. MASCOT: Parameter and state inference under the marginal structured coalescent approximation. Bioinformatics. In press.
<p id="duPlessis2015">du Plessis L., Stadler T. 2015. Getting to the root of epidemic spread with phylodynamic analysis of genomic data. Trends Microbiol. 23:383–386.
<p id="Posada2008">Posada D. 2008. jModelTest: Phylogenetic model averaging. Mol. Biol. Evol. 25:1253–1256.
<p id="Posada2004">Posada D., Buckley T.R. 2004. Model selection and model averaging in phylogenetics: advantages of Akaike information criterion and Bayesian approaches over likelihood ratio tests. Syst. Biol. 53:793–808.
<p id="Pybus2003">Pybus O.G., Drummond A.J., Nakano T., Robertson B.H., Rambaut A. 2003. The epidemiology and iatrogenic transmission of hepatitis C virus in Egypt: a Bayesian coalescent approach. Mol. Biol. Evol. 20:381–387.
<p id="Ripplinger2010">Ripplinger J., Sullivan J. 2010. Assessment of substitution model adequacy using frequentist and Bayesian methods. Mol. Biol. Evol. 27:2790–2803.
<p id="Stadler2012">Stadler T., Kouyos R., von Wyl V., Yerly S., Böni J., Bürgisser P., Klimkait T., Joos B., Rieder P., Xie D. 2012. Estimating the basic reproductive number from viral sequence data. Mol. Biol. Evol. 29:347–357.
<p id="Sullivan2005">Sullivan J., Joyce P. 2005. Model selection in phylogenetics. Annu. Rev. Ecol. Evol. Syst. 36:445–466.
<p id="Vaughan2017">Vaughan T.G., Leventhal G.E., Rasmussen D.A., Drummond A.J., Welch D., Stadler T. 2017. Directly Estimating Epidemic Curves From Genomic Data. bioRxiv.:142570.
<p id="Zie2011">Xie W., Lewis P.O., Fan Y., Kuo L., Chen M.H. 2011. Improving marginal likelihood estimation for Bayesian phylogenetic model selection. Syst. Biol. 60:150–160.
